### DISCUSSION TITLE
How to best migrate my "Neovim from scratch" config to LunarVim's?

### DISCUSSION THREAD

**Discussion by kohane27** [LunarVim/LunarVim#2577] [Q&A] [answered]
Hello guys. Hope you're doing well.

TL;DR: under `~/.config/lvim/`, how do I override LunarVim's default config with the following of mine:

```
.
├── init.lua
├── lua
│   └── user
│       ├── autocommands.lua
│       ├── autopairs.lua
│       ├── bufferline.lua
│       ├── Comment.lua
│       ├── floaterm.lua
│       ├── gitsigns.lua
│       ├── keymaps.lua
│       ├── lightspeed.lua
│       ├── lualine.lua
│       ├── luasnip.lua
│       ├── nvim-tree.lua
│       ├── options.lua
│       ├── sandwitch.lua
│       ├── targets.lua
│       ├── telescope.lua
│       ├── treesitter.lua
```

## Problems

I followed [Neovim from scratch](https://github.com/LunarVim/Neovim-from-scratch) a while back and currently having the above nvim config structure.

My hope is that, I let LunarVim deal with all the LSP-related stuff while maintaining the above [Neovim from scratch](https://github.com/LunarVim/Neovim-from-scratch) directory structure. My question is, **how do I add my stuff onto LunarVim's config?**

I want to retain the above directory structure (putting every plugin's setting to its own file) instead of adding everything to `lvim.plugins` as mentioned in the doc:

```lua
Additional Plugins
lvim.plugins = {
    {"folke/tokyonight.nvim"},
    {
      "folke/trouble.nvim",
      cmd = "TroubleToggle",
    },
}
```

## What I want to achieve

Under `~/.config/lvim/`:

```
.
├── init.lua
├── lua
│   └── user
│       ├── autocommands.lua
│       ├── autopairs.lua
│       ├── bufferline.lua
│       ├── Comment.lua
│       ├── floaterm.lua
│       ├── gitsigns.lua
│       ├── keymaps.lua
│       ├── lightspeed.lua
│       ├── lualine.lua
│       ├── luasnip.lua
│       ├── nvim-tree.lua
│       ├── options.lua
│       ├── sandwitch.lua
│       ├── targets.lua
│       ├── telescope.lua
│       ├── treesitter.lua
```

Then inside `init.lua`:

```lua
require("autocommands.lua")
require("autopairs.lua")
require("bufferline.lua")
require("Comment.lua")
require("floaterm.lua")
require("gitsigns.lua")
require("keymaps.lua")
require("lightspeed.lua")
require("lualine.lua")
require("luasnip.lua")
require("nvim-tree.lua")
require("options.lua")
require("sandwitch.lua")
require("targets.lua")
require("telescope.lua")
require("treesitter.lua")
```

Example of what a lua file looks like (`lightspeed.lua`):

```lua
lightspeed.setup({
    ignore_case = false,
    exit_after_idle_msecs = { unlabeled = 5000, labeled = nil },
    --- s/x ---
    jump_to_unique_chars = { safety_timeout = 800 },
    match_only_the_start_of_same_char_seqs = true,
    force_beacons_into_match_width = false,
    -- Display characters in a custom way in the highlighted matches.
    --- f/t ---
    limit_ft_matches = 4,
    repeat_ft_with_target_char = false,
})
})
```

I understand that some of my config for plugins like `telescope` and `gitsigns` might clash with the [Core Plugins List](https://www.lunarvim.org/plugins/01-core-plugins-list.html). I can manually deal with the conflict. But is the above directory structure possible for "User plugins"?

Any pointer is much appreciated. Thank you!

**Comment by LostInTheLogs** [comment, marked as answer]
Take a look at abzcoding's config, they have it in separate files https://github.com/abzcoding/lvim
